src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp

```diff
@@ -384,28 +428,51 @@ void insert_reorders_in_dir(program& p, const std::map<program_node*, format::ty
         if (fmt_map.count(next) > 0 && fmt_map.at(next) == fmt)
             continue;
 
-        auto next_layout = next->get_output_layout();
-        auto current_layout = node->get_output_layout();
-
-        auto first_layout = travel_direction_wrapper<dir>::first(current_layout, next_layout);
-        auto in_layout = first_layout;
-        auto out_layout = first_layout;
-
-        travel_direction_wrapper<dir>::first(in_layout, out_layout).format = fmt;
+        // We have three (potentially) conflicting information here for format
+        //    node->get_output_layout().format : It is not up-to-date at this moment. It is just the default format (bfyx)
+        //    fmt_map.at(node).format          : It is queried with get_preferred_layout. However, it has only output format.
+        //    node.get_required_input0/output  : If it is valid(!= any), it is up-to-date. It has input format, too.
+        // So the priority is required_input0/output --> fmt_map --> output_layout().format
+
+        auto predecessor = travel_direction_wrapper<dir>::first(node, next);
+        auto successor = travel_direction_wrapper<dir>::second(node, next);
+        auto in_layout = predecessor->get_output_layout();
+        auto out_layout = in_layout;
+        in_layout.format = get_target_output_format(lo, fmt_map, predecessor);
+        auto target_input0_format = get_target_input0_format(lo, fmt_map, successor);
+
+        for (auto& fused_prim : successor->get_fused_primitives()) {
+            // If it is input of fused node, use output layout instead of input layout
+            if (successor->get_dependencies().size() <= fused_prim.dep_start_idx)
+                continue;
```

Comment:
Here it can just break because it has no more deps.

Reply:
OK I will fix it.